src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java

```diff
@@ -1,158 +1,104 @@
-
-    ratingColumn.setCellValueFactory(param -> param.getValue().ratingProperty());
-    ratingColumn.setCellFactory(param -> new StringCell<>(rating -> i18n.number(rating.intValue())));
-
-    contentPane.managedProperty().bind(contentPane.visibleProperty());
-    connectionProgressPane.managedProperty().bind(connectionProgressPane.visibleProperty());
-    connectionProgressPane.visibleProperty().bind(contentPane.visibleProperty().not());
-
-    searchTextField.textProperty().addListener((observable, oldValue, newValue) -> {
-      if (Validator.isInt(newValue)) {
-        ratingTable.scrollTo(Integer.parseInt(newValue) - 1);
-      } else {
-        LeaderboardEntryBean foundPlayer = null;
-        for (LeaderboardEntryBean leaderboardEntry : ratingTable.getItems()) {
-          if (leaderboardEntry.getPlayer().getUsername().toLowerCase().startsWith(newValue.toLowerCase())) {
-            foundPlayer = leaderboardEntry;
-            break;
-          }
-        }
-        if (foundPlayer == null) {
-          for (LeaderboardEntryBean leaderboardEntry : ratingTable.getItems()) {
-            if (leaderboardEntry.getPlayer().getUsername().toLowerCase().contains(newValue.toLowerCase())) {
-              foundPlayer = leaderboardEntry;
-              break;
-            }
-          }
-        }
-        if (foundPlayer != null) {
-          ratingTable.scrollTo(foundPlayer);
-          ratingTable.getSelectionModel().select(foundPlayer);
-        } else {
-          ratingTable.getSelectionModel().select(null);
-        }
+    leaderboardRoot.getSelectionModel().selectedItemProperty().addListener((observable, oldValue, newValue) -> {
+      if (isHandlingEvent) {
+        return;
       }
+      eventBus.post(new OpenLeaderboardEvent(newValue));
     });
   }
 
-  @NotNull
-  private StringConverter<LeaderboardBean> leaderboardStringConverter() {
-    return new StringConverter<>() {
-      @Override
-      public String toString(LeaderboardBean leaderboard) {
-        return i18n.getOrDefault(leaderboard.getTechnicalName(), leaderboard.getNameKey());
-      }
-
-      @Override
-      public LeaderboardBean fromString(String string) {
-        return null;
+  @Override
+  protected void onDisplay(NavigateEvent navigateEvent) {
+    isHandlingEvent = true;
+
+    try {
+      if (navigateEvent instanceof OpenLeaderboardEvent) {
+        OpenLeaderboardEvent event = (OpenLeaderboardEvent) navigateEvent;
+        controllers.forEach(controller -> {
+          if (controller.getRoot().equals(event.getLeagueTab())) {
+            lastTab = controller.getRoot();
+          }
+        });
       }
-    };
-  }
-
-  public void onLeaderboardSelected() {
-    contentPane.setVisible(false);
-    searchTextField.clear();
-    if (usernamesAutoCompletion != null) {
-      usernamesAutoCompletion.dispose();
+      leaderboardRoot.getSelectionModel().select(lastTab);
+    } finally {
+      isHandlingEvent = false;
     }
```

Comment:
I actually have no idea. I mimicked the structure of the onDisplay method of other controllers with subtabs (like the play or replay tab), because I assumed there is a specific reason to do it that way.